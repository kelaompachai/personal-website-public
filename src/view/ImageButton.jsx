import React from 'react';

function ImageButton(props) {
  const { title, link, image } = props;

  return (
    <div className="flex flex-col items-center">
      <h1 className="text-4xl text-wrap text-white max-w-96 text-center font-extralight">{title}</h1>
      <div className="min-h-72 flex justify-around mt-5">
        <a href={link} className={`flex justify-center items-center h-64 w-96 rounded-3xl bg-center bg-no-repeat bg-cover ${image} hover:opacity-80 active:transform active:translate-y-0.5`}></a>
      </div>
    </div>
  );
}

export default ImageButton;
