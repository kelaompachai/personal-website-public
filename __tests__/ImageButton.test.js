import { render, screen } from '@testing-library/react';
import ImageButton from '../src/view/ImageButton';


  it('looks like my image button', () => {
    const container = render(<ImageButton title="testTitle" image="testImage" link="testLink" />);
    expect(container.baseElement).toMatchSnapshot();

    // check for title
    expect(screen.queryByText(/testTitle/)).toBeInTheDocument();

    // check that a element has correct link and image
    const link = screen.getByRole('link');
    expect(link).toHaveAttribute('href', 'testLink');
    expect(link).toHaveClass('testImage');
  });