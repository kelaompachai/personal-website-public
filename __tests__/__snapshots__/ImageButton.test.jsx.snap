// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`looks like my image button 1`] = `
<body>
  <div>
    <div
      class="flex flex-col items-center"
    >
      <h1
        class="text-4xl text-wrap text-white max-w-96 text-center font-extralight"
      >
        testTitle
      </h1>
      <div
        class="min-h-72 flex justify-around mt-5"
      >
        <a
          class="flex justify-center items-center h-64 w-96 rounded-3xl bg-center bg-no-repeat bg-cover testImage hover:opacity-80 active:transform active:translate-y-0.5"
          href="testLink"
        />
      </div>
    </div>
  </div>
</body>
`;
